extensions/ql-vscode/src/databases/ui/db-tree-data-provider.ts

```diff
@@ -48,6 +48,28 @@ export class DbTreeDataProvider
     });
   }
 
+  /**
+   * Updates the selected item and re-renders the tree.
+   * @param selectedItem The item to select.
+   */
+  public async updateSelectedItem(selectedItem: DbTreeViewItem): Promise<void> {
+    const items = await this.getChildren();
```

Comment:
Question: why are we not using `Promise.resolve(this.dbTreeItems)` directly?

Just a nitpick: the getChildren Method states 
```
 /**
   * Called when expanding a node (including the root node).
   * @param node The node to expand.
   * @returns The children of the node.
   */
```
which is not true anymore then

Reply:
Oh very good point! I forgot we had the dbTreeItems available! :)